engine/runtime/src/main/scala/org/enso/compiler/codegen/IrToTruffle.scala

```diff
@@ -703,6 +699,76 @@ class IrToTruffle(
     }
   }
 
+  private def getNestedClass(
+    hostClass: Object,
+    nestedClassName: String
+  ): Option[Object] = {
+    val interop = InteropLibrary.getUncached
+    if (interop.isMemberReadable(hostClass, nestedClassName)) {
+      val nestedClass = interop.readMember(hostClass, nestedClassName)
+      if (interop.isMetaObject(nestedClass)) {
+        return Some(nestedClass)
+      }
+    }
+    None
+  }
+
+  private def getNestedClass(
+    hostClass: Object,
+    nestedClassName: Array[String]
+  ): Option[Object] = {
+    val nestedClasses: ListBuffer[Option[Object]] = ListBuffer(
+      Some(hostClass)
+    )
+    nestedClassName.foreach { name =>
+      nestedClasses.last match {
+        case Some(currentClass) =>
+          getNestedClass(currentClass, name) match {
+            case Some(nestedClass) =>
+              nestedClasses.append(Some(nestedClass))
+            case None =>
+              nestedClasses.append(None)
+          }
+        case None =>
+          nestedClasses.append(None)
+      }
+    }
+    nestedClasses.last
+  }
+
+  /** Tries to lookup a host symbol, i.e., Java class by its fully qualified name.
+    * Can also resolved names of inner classes.
+    * @param name Fully qualified name of the host symbol (Java class).
+    * @return Either host symbol, or throws an exception if not found.
+    */
+  private def tryLookupHostSymbol(name: String): Object = {
+    val items = name.split("\\.")
+    for (idx <- (items.length - 1).to(0) by -1) {
+      val pkgName   = items.take(idx).mkString(".")
+      val className = items(idx)
+      // The rest of the name is interpreted as nested class name
+      val nestedClassPart = items.drop(idx + 1)
+      Try(
+        context.getEnvironment.lookupHostSymbol(pkgName + "." + className)
```

Comment:
This (e.g. Scala code) is an unexpected place for the fix. I think it would be better to "fix" `lookupHostSymbol`. Except my _"Scala problem"_, we want to make sure all **host symbols resolution** behaves the same. I know there is another place that is using `lookupHostSymbol` and now those two would behave inconsistently.

Please replace `context.getEnvironment.lookupHostSymbol` with `EnsoContext.lookupJavaClass` and move all the logic there. Make sure nobody calls `lookupHostSymbol` directly anymore.

Reply:
Done in 74c6cfe5ca1bb7c09727bf5ff5bce5823101b546 - now we have `EnsoContext.lookupJavaClass`.